config/configmap.go

```diff
@@ -32,8 +33,7 @@ const (
 
 // MapConverterFunc is a converter function for the config.Map that allows distributions
 // (in the future components as well) to build backwards compatible config converters.
-// TODO: Consider to add a context as the first argument.
-type MapConverterFunc func(*Map) error
+type MapConverterFunc func(context.Context, *Map) error
```

Comment:
@tigrannajaryan what do you think about `ConvertFunc` as the name?

Reply:
I think `MapConverterFunc` is appropriate, otherwise it might be unclear what it operates on.